### PR TITLE
[azure-arm-rest-v2, az-blobstorage-provider] Added preview version for Node 10 migration #13909

### PR DESCRIPTION
**Pull request by sdobrodeev**
**Task name**: azure-arm-rest-v2, az-blobstorage-provider

**Description**: 

- Updated packages for Node 10 migration.
- Updated packages versions

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

File: common-npm-packages/azure-arm-rest-v2/package.json [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
-    "@types/q": "1.0.7"
+    "azure-pipelines-task-lib": "^3.0.6-preview.0",
+    "@types/mocha": "^5.2.7",
+    "@types/node": "^10.17.0",
+    "@types/q": "1.5.4"
   },
   "devDependencies": {
-    "typescript": "2.3.4",
+    "typescript": "4.0.2",
     "shelljs": "0.8.4"
```
Comment (DaniilShmelev) [inline common-npm-packages/azure-arm-rest-v2/package.json:26]: Tasks can only have typescript in devDependencies, does the same limitation apply to common npm packages?

Comment (sdobrodeev) [reply common-npm-packages/azure-arm-rest-v2/package.json:26]: I think yes.

**Review by EzzhevNikita — APPROVED** [review]

File: common-npm-packages/az-blobstorage-provider-v2/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azp-tasks-az-blobstorage-provider-v2",
-  "version": "1.0.0",
+  "version": "1.0.0-preview.0",
```
Comment (egor-bryzgalov) [inline common-npm-packages/az-blobstorage-provider-v2/package.json:3]: I propose to increase major version.

File: common-npm-packages/azure-arm-rest-v2/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "azure-pipelines-tasks-azure-arm-rest-v2",
-  "version": "1.177.0",
+  "version": "1.177.0-preview.0",
```
Comment (egor-bryzgalov) [inline common-npm-packages/azure-arm-rest-v2/package.json:3]: Let's create `2.0.0-preview.0`

**Review by egor-bryzgalov — APPROVED** [review]
Could you please take a look at the comments? Otherwise LGTM.

**Pull request merged**